activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -568,9 +568,12 @@ def translate_exception_class(e, sql)
         end
 
         def log(sql, name = "SQL", binds = [], type_casted_binds = [], statement_name = nil) # :doc:
+          resolver = ConnectionSpecification::Resolver.new(Base.configurations)
+
           @instrumenter.instrument(
             "sql.active_record",
             sql:               sql,
+            spec_name:         resolver.spec(@config).name,
```

Comment:
Can't we access this using `poll.spec.name` since we are already inside the connection? Of course when the `pool` is already set.